specification/purview/data-plane/Azure.Analytics.Purview.DataMap/preview/2024-03-01-preview/purviewdatamap.json

```diff
@@ -6866,11 +6941,140 @@
       "title": "SuggestRequest",
       "description": "The query of suggest request."
     },
+    "NavigationRequest": {
+      "type": "object",
+      "description": "The request payload of Navigation API.",
+      "properties": {
+        "navigationMode": {
+          "$ref": "#/definitions/NavigationMode"
+        },
+        "itemPath": {
+          "$ref": "#/definitions/ItemPath"
+        },
+        "properties": {
+          "type": "object",
+          "description": "The additional properties of the item.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "includeNextLevelAssetCount": {
+          "type": "boolean",
+          "description": "Whether to return the next level asset count."
+        }
+      }
+    },
+    "NavigationMode": {
+      "type": "string",
+      "description": "The enum of navigation mode.",
+      "enum": [
+        "assetType",
+        "azureResourceHierarchy"
+      ],
+      "x-ms-enum": {
+        "name": "NavigationMode",
+        "modelAsString": true
+      }
+    },
+    "ItemPath": {
+      "type": "object",
+      "properties": {
+        "path": {
+          "type": "string",
+          "description": "The identifier of the item. The path starts with an empty string or a single slash. To navigate further, you need to concatenate with a slash to form the full itemPath using the previous navigation response's relative item path."
+        },
+        "extendedProperties": {
+          "type": "object",
+          "description": "The extended properties of the itemPath are typically obtained from the last navigation response. While not mandatory to provide, including them can enhance performance. Otherwise, there may be some impact on performance.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      },
+      "description": "The identifier of navigation request."
+    },
+    "NavigationResult": {
+      "type": "object",
+      "properties": {
+        "items": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NavigationElement"
+          }
+        },
+        "continuationToken": {
+          "type": "string",
+          "description": "The token used to get next batch of data. Absent if there's no more data."
+        }
+      },
+      "description": "The response payload of the Navigation API."
+    },
+    "NavigationElement": {
+      "type": "object",
+      "description": "The item payload of the NavigationResult.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the item."
+        },
+        "type": {
+          "type": "string",
+          "description": "The type name of the item. Eg. EntityType."
+        },
```

Comment:
Are any of these properties required? I would think at least `name` would be, if not `type` and maybe even `itemPath`.

Reply:
Even if it's response-only (read-only response properties), marking these `required` informs code generators what type to use for the generated model property.

Reply:
updated the required fields